lib/utils.go

```diff
@@ -1304,30 +1309,29 @@ func testJailParseInvalid(t *testing.T) bool {
 	return true
 }
 
-func testJailInit(t *testing.T) bool {
-	initCode := `
-	var _status_catalog = {
-		foo: 'bar'
-	};
-	`
-	InitJail(C.CString(initCode))
-
-	extraCode := `
-	var extraFunc = function (x) {
-	  return x * x;
-	};
-	`
-	rawResponse := CreateAndInitCell(C.CString("CHAT_ID_INIT_TEST"), C.CString(extraCode))
+func checkExportedFunction(jailFunc jailFunc, chatName, jsCode, expectedResponse string, t *testing.T) bool {
+	rawResponse := jailFunc.FuncRef(C.CString(chatName), C.CString(jsCode))
 	parsedResponse := C.GoString(rawResponse)
 
-	expectedResponse := `{"result": {"foo":"bar"}}`
-
 	if !reflect.DeepEqual(expectedResponse, parsedResponse) {
-		t.Error("expected output not returned from jail.CreateAndInitCell()")
+		t.Errorf("expected output not returned from %s (actual: %s, expected %s)", jailFunc.Name, parsedResponse, expectedResponse)
+		return false
+	}
+	return true
+}
+
+func testJailInit(t *testing.T) bool {
+	InitJail(C.CString(`var _status_catalog = { foo: 'bar' };`))
+
+	extraCode := `var extraFunc = function (x) { return x * x; }; extraFunc(2);`
+	if !checkExportedFunction(jailFunc{CreateAndInitCell, "jail.CreateAndInitCell()"}, "CHAT_ID_INIT_TEST", extraCode, `{"result": 4}`, t) {
```

Comment:
I am not sure if this helper is needed. The biggest assumption of it is that it all jail functions have the same signature which may be true now but won't have to be tomorrow.

Maybe we could just replace `reflect.DeepEqual` call with `require.Equal`. It could look like this:

```go
response := CreateAndInitCell(
  C.CString("CHAT_ID_INIT_TEST"), 
  C.CString(`var extraFunc = function (x) { return x * x; }; extraFunc(2);`),
)
require.Equal(t, `{"result": 4}`, C.GoString(response))
```

I think it's not too much code and it's pretty easy to understand.

Reply:
I agree with you, the whole idea of this refactoring was to make Code Climate happy.
It currently has these checks for 25 LoC per function and for max 4 return statements.

I think these limits make sense for the code itself, but maybe they shouldn't be that strict with test methods.

Reply:
I am not sure we should care about Code Climate now. This whole file needs heavy refactoring and using `require.Equal` should make it happier in terms of different code paths anyway.

Reply:
Ah, 👌 
I thought it would block the merging process.